developer-process/code-guidelines.md

```diff
@@ -0,0 +1,148 @@
+
+
+## Naming
+
+* camelCase for identifiers
+* PascalCase for classes
+* dash-case for filenames
+* file names and class names are carefully matched
+* verbNoun naming for functions and methods: 'handleEvent', not 'handle'.
+* always type out identifiers, no one letter of abbreviations. `array.map(element => element + 1)` instead of `array.map(e => e + 1)`;
+
+As always, exceptions OK where it makes sense (e.g. `AnimationLoop.start` breaks the verbNoun rule).
+
+
+## Logging
+
+We use [probe.gl](https://uber-web.github.io/probe.gl) for all information printed to the console. The minimalist library allows us to manage formatting, repeated messages and log priority in one place.
+
+* API deprecation and removal
+* Warnings
+* Errors
+* Debug logs
+
+
+## asserts
+
+We use `assert`s to check that proper parameters are passed.
+
+We typically do not include any helpful messages to the programmer. These bloat the library and provide little additional value beyond what the developer can glean from his program breaking in the assert.
+
+Exceptions can be made if the particular condition being asserted against is a particular pitfall for programmers and/or dependent on input data (e.g. the failure to conform to the `[lng, lat]` convention in several of our libraries).
+
+Our frameworks have local definitions of `assert`, to avoid bundlers injecting the bloated node polyfill.
+
+
+### Why do we use asserts?
+
+The problem that (most of) our asserts are targeting:
+
+* Asserts move the detection of errors related to apps passing incorrect parameters from the internals of our libraries to the API surface of these libraries.
+* This means that app developers will get an exception immediately in code that he is familiar with (his invocation of the library API) rather than later getting a mystic failure in the internal of our library.
+* A number of asserts have been added after developers "waster" time debugging issues just to trace them back to the app sending in a bad parameter.
+
+
+### Why do we not include error messages in asserts?
+
+We removed the messages since we felt that they added bundle size bloat to our already "oversized" libraries, but limited additional debugging value.
+
+* One reason that messages add limited value is that as soon as an assert happens, it is trivial to turn on "break on exceptions" and rerun.
+* When the debugger stops at the assert in the non-minified version, showing clearly which param is wrong in which API call, a human readable error message adds little.
+
+
+### But, asserts are not useful with minified version?
+
+While there are rare cases when an error only happens in the minified version, there are so many other limitations with debugging in minified versions.
+
+
+### Transpiler
+
+We use babel-preset-env to transpile code for publishing.
+
+We use buble or babel to transpile examples.
```

Comment:
This is not standardized, across frameworks but my approach is to not transpile examples at all (just bundle them).

I don't think we use buble anymore, although there might be a few straggler examples.